From 3aaee0415269cf093b218ff8bb32b800a38b4834 Mon Sep 17 00:00:00 2001
From: John <john@primaryautotrd.com>
Date: Thu, 1 Oct 2026 23:19:00 +0300
Subject: [PATCH] gppservice: allow binder calls to gmscore_app and system_app

GMS runs in its own gmscore_app domain (denied during Cast screen mirroring,
gppservice calling back into the MediaProjection client); system_app for RedMagicControl.

Change-Id: Ib720df5e15aa30074dbecae09b7c48dc49ef713e
---
 generic/private/gppservice.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/generic/private/gppservice.te b/generic/private/gppservice.te
index 9825d0c0..e4e89369 100644
--- a/generic/private/gppservice.te
+++ b/generic/private/gppservice.te
@@ -24,6 +24,10 @@ binder_call(vendor_gppservice, surfaceflinger);
 binder_call(vendor_gppservice, platform_app);
 binder_call(vendor_gppservice, priv_app);
 binder_call(vendor_gppservice, untrusted_app_all);
+# GMS runs in its own gmscore_app domain (denied 2026-09-07 during Cast screen mirroring,
+# gppservice calling back into the MediaProjection client); system_app for RedMagicControl.
+binder_call(vendor_gppservice, gmscore_app);
+binder_call(vendor_gppservice, system_app);
 
 allow vendor_gppservice gpu_device:chr_file rw_file_perms;
 
-- 
2.55.0

